src/wl_init.c

```diff
@@ -1078,7 +1081,10 @@ int _glfwPlatformInit(void)
 
     if (_glfw.wl.pointer && _glfw.wl.shm)
     {
-        _glfw.wl.cursorTheme = wl_cursor_theme_load(NULL, 32, _glfw.wl.shm);
+        cursorTheme = getenv("XCURSOR_THEME");
+        cursorSize = getenv("XCURSOR_SIZE") ? atoi(getenv("XCURSOR_SIZE")) : 32;
```

Comment:
There is a race condition here, please load the env var in a variable, and only then do the check for `NULL` and the conversion.

Also, please use `strtol()` instead of `atoi()` in order to check for errors.